### ISSUE TITLE
Improve tedge_apt_plugin error messages

### ISSUE THREAD

**Issue by didier-wenzek** [thin-edge/thin-edge.io#1183]
When used to install a Debian package previously downloaded, the `tedge_apt_plugin` command checks that the file given on the command line is actually the expected one. This is nice but the error message is misleading.

When the command is run with a correct Debian package but using a wrong package name, `tedge_apt_plugin` complains that the file cannot be parsed, while the error is that the actual name of the package is not the expected one.

For instance, given a file `/tmp/thinml_3964` which content is a Debian package named `zementis-thinml-3964`:

```
$ dpkg -I /tmp/thinml_3964 | grep Package
Package: zementis-thinml-3964
```

And installing it under the wrong name `thinml-3964`, triggers a misleading error message:

```
$ sudo /etc/tedge/sm-plugins/apt install thinml-3964 --file /tmp/thinml_3964
ERROR: Parsing Debian package failed for `/tmp/thinml_3964`
```

What I would expect is an error telling the mismatch between the name given on the cloud and the actual name of the package:

```
$ sudo /etc/tedge/sm-plugins/apt install thinml-3964 --file /tmp/thinml_3964
ERROR: Debian package name mismatch
    expected Package: thinml-3964
    actual      Package: zementis-thinml-3964
```

I expect the same kind of error message when the version given on the cloud doesn't match the actual version of the package:

```
$ dpkg -I /tmp/thinml_3964 | grep Version
Version: 0.0.1-1

$ sudo /etc/tedge/sm-plugins/apt install zementis-thinml-3964 --file /tmp/thinml_3964 --module-version 1.0
ERROR: Debian package name mismatch
    expected Version: 1.0
    actual      Version: 0.0.1-1
```

The parsing error should be displayed only when the file is actually not a Debian package.
Having the root error would be a plus.

```
$ echo "Not a debian package" >/tmp/foo.deb

$ sudo /etc/tedge/sm-plugins/apt install thinml-3964 --file /tmp/foo.deb
ERROR: Parsing Debian package failed for `/tmp/foo.deb`
dpkg-deb: error: '/tmp/foo.deb' is not a Debian format archive
```

**Comment by PradeepKiruvale** [comment]
Steps to test:

- Install the latest version (0.7.4) thin-edge.io, this must install `tedge_apt_plugin` use the below command line
      `curl -fsSL https://raw.githubusercontent.com/thin-edge/thin-edge.io/main/get-thin-edge_io.sh | sudo sh -s 0.7.4`
- Download the rolldice Debian package
    ` wget  http://ports.ubuntu.com/pool/universe/r/rolldice/rolldice_1.16-1build1_arm64.deb`


- First test case with the wrong package name
Use the below command line to invoke the apt plugin
`sudo /etc/tedge/sm-plugins/apt install thinml-3964 --file ./rolldice_1.16-1build1_arm64.deb`
Should receive the below error message
`ERROR: Validation of ./rolldice_1.16-1build1_arm64.deb metadata failed, expected value for the Package is  rolldice, but provided  thinml-3964`

- Second test case with the wrong version
Use the below command to invoke the apt plugin with wrong version
`sudo /etc/tedge/sm-plugins/apt install thinml-3964 --file ./rolldice_1.16-1build1_arm64.deb --module-version 1.0`
The expected error message should be as below
`ERROR: Validation of ./rolldice_1.16-1build1_arm64.deb metadata failed, expected value for the Version is  1.16-1build1, but provided  1.0`

- Third test case with the wrong type of package 
Create a wrong debian package as below
`echo "Not a debian package" >/tmp/foo.deb`
Try to install the package using below command
`sudo /etc/tedge/sm-plugins/apt install thinml-3964 --file /tmp/foo.deb`
The expected error message should look as below
`ERROR: Parsing Debian package failed for `/tmp/foo.deb`, Error: dpkg-deb: error: '/tmp/foo.deb' is not a Debian format archive`

**Comment by PradeepKiruvale** [comment]
Resolved with #1341
